core/runtime/ops.rs

```diff
@@ -901,15 +903,42 @@ mod tests {
     *state.borrow_mut() = value;
   }
 
+  #[op2(core, fast)]
+  pub fn op_state_mut_attr(#[state] value: &mut u32, new_value: u32) -> u32 {
+    let old_value = *value;
+    *value = new_value;
+    old_value
+  }
+
+  #[op2(core, fast)]
+  pub fn op_state_multi_attr(
+    #[state] value32: &u32,
+    #[state] value16: &u16,
+    #[state] value8: Option<&u8>,
```

Comment:
This feels very magical... I'm not sure how I feel about it. Is there a way to drop this borrow somehow?

Reply:
OpState is borrowed during the entire time we're in these functions (regardless of whether we're doing this with the helpers or directly against OpState) -- the `borrow` on `GothamState` is a fake borrow that is more like a "deref the item with this key".

Technically you can control the lifetime of the borrow using Rc<RefCell<OpState>> but usually we just assume that OpState is locked during the entire run of a sync function.

Reply:
This just desugars to:

```
let a: &A = opstate.borrow().borrow::<A>()
let b: &B = opstate.borrow().borrow::<B>()
```

Where it can be extremely useful is if we tweak the state to allow for multiple mutable borrows that might not be as ergonomic as this, or if we decide to change the state interface.